pennylane/math/quantum.py

```diff
@@ -606,3 +606,158 @@ def _compute_mutual_info(
     )
 
     return vn_entropy_1 + vn_entropy_2 - vn_entropy_12
+
+
+def fidelity(state0, state1, check_state=False, c_dtype="complex128"):
+    r"""Compute the fidelity for two states (a state can be a state vector or a density matrix) acting on quantum
+    systems with the same size. For two pure states, the fidelity corresponds to the squared overlap. For a pure state
+    and a mixed state, it corresponds to the squared expectation of the mixed state in the pure state. Finally for two
+    mixed states, it is defined by the last formula:
+
+    .. math::

```

Comment:
Can you do case1 - formula, case 2 - formula, case 3 - formula or enumerate the formulas? Maybe by hand by adding something like `\vspace \text{ (1)}` after each one

Reply:
Thank you 👍